### PR TITLE
trap SIGINT to suppress noisy stack traces

### PR DESCRIPTION
**Pull request by salsferrazza**
Control-C'ing out of a transcoder process run in the foreground exits with a very noisy stack trace that inhibits immediate diagnostics of the last few messages. This CL traps SIGINT and exists more gracefully. It also provides the basic entry point for #13 and #111.

### CODE REVIEW

**Review by mservidio — DISMISSED** [review]

**Review by t8dogg — COMMENTED** [review]
Useful and comprehensible, LGTM

**Review by t8dogg — APPROVED** [review]

**Review by t8dogg — APPROVED** [review]

**Comment by mservidio** [comment]
@salsferrazza needs some linting clean up, I touched up a couple of the issues

**Review by t8dogg — APPROVED** [review]
The linting LGTM @mservidio

File: transcoder/message/MessageParser.py [hunk 1]
```diff
@@ -127,25 +130,27 @@ def setup_handlers(self, message_handlers: str):
...
 
-            self.process_data(source)
+            self.process_data()
 
         if self.output_manager is not None:
             self.output_manager.wait_for_completion()
 
+        self.summary()
+
+    def summary(self):
```
Comment (mservidio) [inline transcoder/message/MessageParser.py:150]: Should probably rename to print_summary or log_summary?

File: transcoder/main.py [hunk 2]
```diff
@@ -37,10 +39,13 @@
 from transcoder.source import all_source_identifiers
 
 script_dir = os.path.dirname(__file__)
+message_parser: MessageParser = None
 
 
 def main():
     """main entry point for Datacast Transcoder"""
+
+    global message_parser
```
Comment (mservidio) [inline transcoder/main.py:48]: The linter doesn't like using global. It appears that signal can be wired up to instances: https://stackoverflow.com/questions/34786524/signal-handler-inside-a-class

**Review by mservidio — CHANGES_REQUESTED** [review]

**Comment by mservidio** [comment]
One other thought, the logic in main.py will only be a feature add for cli usage, but not library usage. Moving this into the MessageParser instance may be more beneficial.

**Comment by mservidio** [comment]
I'm going to mark this WIP, and raise a separate PR trapping from MessageParser.

**Comment by mservidio** [comment]
@salsferrazza re-factored a little, take a look...

**Review by mservidio — APPROVED** [review]

**Pull request merged**